rpy/robjects/pandas2ri.py

```diff
@@ -141,13 +142,15 @@ def py2rpy_pandasseries(obj):
         }[homogeneous_type](obj)
     elif obj.dtype.name in integer_array_types:
         if not obj.dtype.numpy_dtype.isnative:
-            raise(ValueError('Cannot pass numpy arrays with non-native byte orders at the moment.'))
+            raise(ValueError('Cannot pass numpy arrays with non-native byte'
+                             ' orders at the moment.'))
         if obj.dtype.kind == 'i':
             vec = IntVector(numpy.ravel(obj, order='F'))
             dim = IntVector(obj.shape)
             res = rinterface.baseenv['array'](vec, dim=dim)
         elif obj.dtype.kind == 'u':
-            raise (ValueError('Cannot convert numpy array of unsigned values -- R does not have unsigned integers.'))
+            raise (ValueError('Cannot convert numpy array of unsigned values'
```

Comment:
Here as well. I think that this conversion should be handled by the numpy converter as this is not pandas-specific.

Reply:
Technically unsigned integers in Python can be converted to R's (signed) integers if the number of the bits used for the Python integer is lower than the number of bits use in R integers.

Reply:
also from numpy code: https://github.com/rpy2/rpy2/blob/master/rpy/robjects/numpy2ri.py#L71

Reply:
Previously you suggested moving this code to the pandas module. Now you want it back in the numpy module?

I wasnt sure how to do parsing in the pandas module without copying the relevant code.

Reply:
> Previously you suggested moving this code to the pandas module. Now you want it back in the numpy module?

I can't rule rule that I misunderstood your code and intent, not to mention that own code did not benefit from code review so there is a fair chance that I commenting on what I am doing elsewhere ;-).

The idea is to try avoiding code-duplication. If reach a point where we are sure that this is a numpy array, the next conversion steps should be delegated to the numpy converter.

Reply:
I understand and I think code duplication is bad. But to make `rpy2` handle both `numpy` and `pandas` data properly it would require a bit of restructuring of the `numpy` module.